test/e2e_node/node_shutdown_linux_test.go

```diff
@@ -197,9 +198,167 @@ var _ = SIGDescribe("GracefulNodeShutdown [Serial] [NodeAlphaFeature:GracefulNod
+			customClassC = getPriorityClass("custom-class-c", 1000)
+		)
+
+		tempSetCurrentKubeletConfig(f, func(initialConfig *kubeletconfig.KubeletConfiguration) {
+			initialConfig.FeatureGates = map[string]bool{
+				string(features.GracefulNodeShutdown):             true,
+				string(features.PodPriorityBasedGracefulShutdown): true,
+			}
+			initialConfig.PodPriorityShutdownGracePeriods = []kubeletconfig.PodPriorityShutdownGracePeriod{
+				{
+					Priority:                   scheduling.SystemCriticalPriority,
+					ShutdownGracePeriodSeconds: int64(podStatusUpdateTimeout / time.Second),
+				},
+				{
+					Priority:                   customClassA.Value,
+					ShutdownGracePeriodSeconds: int64(podStatusUpdateTimeout / time.Second),
+				},
+				{
+					Priority:                   customClassB.Value,
+					ShutdownGracePeriodSeconds: int64(podStatusUpdateTimeout / time.Second),
+				},
+				{
+					Priority:                   customClassC.Value,
+					ShutdownGracePeriodSeconds: int64(podStatusUpdateTimeout / time.Second),
+				},
+				{
+					Priority:                   scheduling.DefaultPriorityWhenNoDefaultClassExists,
+					ShutdownGracePeriodSeconds: int64(podStatusUpdateTimeout / time.Second),
+				},
+			}
+
+		})
+
+		ginkgo.BeforeEach(func() {
+			ginkgo.By("Wait for the node to be ready")
+			waitForNodeReady()
+
+			for _, customClass := range []*schedulingv1.PriorityClass{customClassA, customClassB, customClassC} {
+				_, err := f.ClientSet.SchedulingV1().PriorityClasses().Create(context.Background(), customClass, metav1.CreateOptions{})
+				framework.ExpectNoError(err)
+			}
+		})
+
+		ginkgo.AfterEach(func() {
+			ginkgo.By("Emitting Shutdown false signal; cancelling the shutdown")
+			err := emitSignalPrepareForShutdown(false)
+			framework.ExpectNoError(err)
+		})
+
+		ginkgo.It("should be able to gracefully shutdown pods with various grace periods", func() {
+			nodeName := getNodeName(f)
+			nodeSelector := fields.Set{
+				"spec.nodeName": nodeName,
+			}.AsSelector().String()
+
+			// Define test pods
+			pods := []*v1.Pod{
+				getGracePeriodOverrideTestPod("period-120", nodeName, 120, ""),
+				getGracePeriodOverrideTestPod("period-5", nodeName, 5, ""),
+				getGracePeriodOverrideTestPod("period-c-120", nodeName, 120, customClassC.Name),
+				getGracePeriodOverrideTestPod("period-c-5", nodeName, 5, customClassC.Name),
+				getGracePeriodOverrideTestPod("period-b-120", nodeName, 120, customClassB.Name),
+				getGracePeriodOverrideTestPod("period-b-5", nodeName, 5, customClassB.Name),
+				getGracePeriodOverrideTestPod("period-a-120", nodeName, 120, customClassA.Name),
+				getGracePeriodOverrideTestPod("period-a-5", nodeName, 5, customClassA.Name),
+				getGracePeriodOverrideTestPod("period-critical-120", nodeName, 120, scheduling.SystemNodeCritical),
+				getGracePeriodOverrideTestPod("period-critical-5", nodeName, 5, scheduling.SystemNodeCritical),
+			}
+
+			// Expected down steps
+			downSteps := [][]int{
```

Comment:
this is a bit confusing, maybe it would be more clear to define the pods with specific names above and then refer to them here instead? 

Otherwise let's leave a comment to make it clear that the values here refer to the index in the pods slice.